java-client/src/main/java/org/opensearch/client/transport/rest_client/RestClientTransport.java

```diff
@@ -370,7 +370,6 @@ private <ResponseT> ResponseT decodeResponse(
             InputStream content = null;
             if (entity != null) {
                 // We may have to replay it.
-                entity = new BufferedHttpEntity(entity);
```

Comment:
Removing unnecessary buffering since response is converted to bytes now, always (https://github.com/opensearch-project/opensearch-java/pull/984/files#diff-0eec44b1849b65152d79b3673d9ebb77442080c306919c5e931506a6aa0a31f0R98)